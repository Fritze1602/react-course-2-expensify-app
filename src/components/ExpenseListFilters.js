import React from 'react';
import {connect} from 'react-redux';
import {setTextFilter, sortByAmount, sortByDate, setStartDate, setEndDate} from '../actions/filters';
import {DateRangePicker} from 'react-dates'

class ExpenseListFilters extends React.Component{
    state = {
        calenderFocused: null
    }
    onDatesChange = ({startDate, endDate}) => {
        this.props.dispatch(setStartDate(startDate));
        this.props.dispatch(setEndDate(endDate))
    }
    onFocusChange = (calenderFocused) => {
        this.setState(()=>({
            calenderFocused
        }))
    }
    render () {
        return (
            <div>
                <input 
                    type="text" 
                    value={this.props.filters.text} 
                    onChange={(e)=>{
                        this.props.dispatch(setTextFilter(e.target.value));
                }}/>
                <select 
                    value={this.props.filters.sortBy}
                    onChange={(e) => {
                        e.target.value === 'amount' ? this.props.dispatch(sortByAmount()) : this.props.dispatch(sortByDate());
                }}>
                    <option value="date">Date</option>
                    <option value="amount">Amount</option>
                </select>
                <DateRangePicker 
                    startDate={this.props.filters.startDate}
                    startDateId="your_unique_start_date_id" 
                    endDate={this.props.filters.endDate}
                    endDateId="your_unique_end_date_id" 
                    onDatesChange={this.onDatesChange}
                    focusedInput={this.state.calenderFocused}
                    onFocusChange={this.onFocusChange}
                    showClearDates={true}
                    numberOfMonths={1}
                    isOutsideRange={()=>false}
                    minimumNights={0}

                />
            </div>
        )
    }
}


const mapStateToProps = (state) => {
    return {
        filters:state.filters
    }
}

export default connect(mapStateToProps)(ExpenseListFilters);