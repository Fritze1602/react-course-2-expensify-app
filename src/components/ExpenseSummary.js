import React from 'react';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';
import numeral from 'numeral'
import {connect} from 'react-redux';

export const ExpenseSummary = ({expensesCount, expensesTotal}) => {
    const expensesWord = expensesCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal/100).format('$0,0.00');
    return (
        <div>
            <p>Viewing {expensesCount} {expensesWord} with amount of {formattedExpensesTotal}.</p>
        </div>
    )
}

const mapStateToProps = (state)=>{
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expensesCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    }
}

export default connect(mapStateToProps)(ExpenseSummary);

