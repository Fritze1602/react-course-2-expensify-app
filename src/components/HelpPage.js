import React from 'react';

const HelpPage = () => (
    <div>This is my new Help Page</div>
);

export default HelpPage;