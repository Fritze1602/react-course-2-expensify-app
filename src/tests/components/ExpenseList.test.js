import React from 'react';
import {shallow} from 'enzyme';
import expenses from '../fixtures/expenses'

import {ExpenseList} from '../../components/ExpenseList';

test('Should render List with expenses (fixtures)', () => {
    const wrapper = shallow(<ExpenseList expenses={expenses}/>);
    expect(wrapper).toMatchSnapshot();
})

test('Should render Expense List withe empty Message', () => {
    const wrapper = shallow(<ExpenseList expenses={[]}/>);
    expect(wrapper).toMatchSnapshot();
})