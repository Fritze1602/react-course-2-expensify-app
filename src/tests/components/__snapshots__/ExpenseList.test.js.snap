// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Expense List withe empty Message 1`] = `
<div
  className="content-container"
>
  <div
    className="list-header"
  >
    <div
      className="show-for-mobile"
    >
      Expenses
    </div>
    <div
      className="show-for-desktop"
    >
      Expense
    </div>
    <div
      className="show-for-desktop"
    >
      Amount
    </div>
  </div>
  <div
    className="list-body"
  >
    <p
      class="list-item list-item--message"
    >
      No expenses yet
    </p>
  </div>
</div>
`;

exports[`Should render List with expenses (fixtures) 1`] = `
<div
  className="content-container"
>
  <div
    className="list-header"
  >
    <div
      className="show-for-mobile"
    >
      Expenses
    </div>
    <div
      className="show-for-desktop"
    >
      Expense
    </div>
    <div
      className="show-for-desktop"
    >
      Amount
    </div>
  </div>
  <div
    className="list-body"
  >
    <ExpenseListItem
      amount={195}
      createdAt={0}
      description="Gum"
      id="1"
      key="1"
      note=""
    />
    <ExpenseListItem
      amount={295000}
      createdAt={-345600000}
      description="Rent"
      id="2"
      key="2"
      note=""
    />
    <ExpenseListItem
      amount={4500}
      createdAt={345600000}
      description="Credit Card"
      id="3"
      key="3"
      note=""
    />
  </div>
</div>
`;
